### DISCUSSION TITLE
Backwards incompatible Changes on Feb. 25

### DISCUSSION THREAD

**Discussion by berchn** [tree-sitter/tree-sitter#951] [Q&A] [answered]
Commit 774ae5e3d5915dbdcca52e0abf54c3a6023cca3e introduced some backwards incompatible changes that are breaking a CFFI wrapper I'm using--`ts_parser_set_language` no longer returns the expected value. What changes have been made that might cause this?

**Comment by maxbrunsfeld** [comment]
Yeah, this was intentional. Sorry for the confusion though.

This coming week I will publish a new version of the CLI that generates parsers with the new binary interface, so that they’ll be usable with the latest version of the library.

**Reply by maxbrunsfeld** [reply, marked as answer]
For a long time, we’ve able to maintain backward compatibility during changes, but there were some issues that required a breaking change in order to adress.

**Reply by berchn** [reply]
Alright, thanks. I had tried rebuilding everything--CLI + language modules--and nothing had worked.
